include/Gaffer/Monitor.h

```diff
@@ -83,6 +83,21 @@ class GAFFER_API Monitor : public IECore::RefCounted
 		/// on this thread.
 		static const MonitorSet &current();
 
+		/// Must return true if disallowCacheRead will ever return true from this Monitor
+		/// ( for the given type )
+		virtual bool mightDisallowCacheRead( ThreadState::CacheType type )
+		{
+			return false;
+		}
```

Comment:
In this case, to my view, looking at the header as "information that will be useful to implementor", the fact that these functions default to returning false seems like a useful thing to advertise.  But since they're virtual, there's no performance benefit.  Moved to cpp under protest.